sdk/core/Azure.Core/src/DynamicData/MutableJsonDocument.ChangeTracker.cs

```diff
@@ -135,6 +141,70 @@ internal IEnumerable<MutableJsonChange> GetRemovedProperties(string path, int hi
                 }
             }
 
+            internal MutableJsonChange? GetFirstMergePatchChange(ReadOnlySpan<char> rootPath, out int maxPathLength)
+            {
+                // This method gets the first change from the list in sorted order by path
+                // It also returns the max path length of changes on the list.
+
+                maxPathLength = -1;
+
+                if (_changes == null)
```

Comment:
Maybe if `_changes is null` and you won't need `_changes!.Count` below? Just a thought. Seems odd you needed that on line 158.